File: scripts/test_solutions/543_diameter_binary_tree.py
def diameterOfBinaryTree(root):
    diameter = [0]

    def depth(node):
        if not node:
            return 0
        left = depth(node.left)
        right = depth(node.right)
        diameter[0] = max(diameter[0], left + right)
        return 1 + max(left, right)

    depth(root)
    return diameter[0]

File: scripts/test_solutions/543_diameter_binary_tree.js
var diameterOfBinaryTree = function(root) {
    let diameter = 0;
    function depth(node) {
        if (!node) return 0;
        const left = depth(node.left), right = depth(node.right);
        diameter = Math.max(diameter, left + right);
        return 1 + Math.max(left, right);
    }
    depth(root);
    return diameter;
};
